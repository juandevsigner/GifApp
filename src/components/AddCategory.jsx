import { useState } from "react";
import Proptypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = e => {
    e.preventDefault();
    if (inputValue.trim().length <= 1) return;
    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form aria-label="form" onSubmit={onSubmit}>
      <input
        type="text"
        value={inputValue}
        onChange={onInputChange}
        placeholder="Search Gifs"
      />
    </form>
  );
};

AddCategory.proptypes = {
  onNewCategory: Proptypes.func.isRequired,
};
