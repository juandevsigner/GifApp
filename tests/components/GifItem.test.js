import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"


describe('Test GitItem', () =>{
    const title ='Saitama'
    const url = 'https://saitama.com/'

    it('Snapshot vs Component', () =>{
        const {container} = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })

     it('should have class "card', () =>{
       const {container} = render(<GifItem title={title} url={url} />)
       expect(container.querySelector('.card').className).toBeTruthy()

    })

    it('Should be image with correct url', () =>{
        render(<GifItem title={title} url={url} />)
        expect(screen.getByRole('img').src).toBe(url)
        expect(screen.getByRole('img').alt).toBe(title)
    })

    it('should show correct text', () =>{
        render(<GifItem title={title} url={url} />)
        expect(screen.getByText(title).innerHTML).toBe(title)
    })


})