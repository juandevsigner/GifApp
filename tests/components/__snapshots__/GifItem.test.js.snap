// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GitItem Snapshot vs Component 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://saitama.com/"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
